### PR TITLE
Revise HPA documentation for stable HPA v2 API

### PR DESCRIPTION
**Pull request by josephburnett**
Document the new HPA v2 stable API which replaces HPA v2beta2.

KEP: https://github.com/kubernetes/enhancements/blob/master/keps/sig-autoscaling/2702-graduate-hpa-api-to-GA/README.md

### CODE REVIEW

**Comment by chrisnegus** [comment]
@josephburnett I see you have already updated some files and don't have a hold on this PR. Is it ready to review or do you still have more work to do on it?

**Review by sftim — COMMENTED** [review]
Hi. Here's some feedback on this PR, which at the time of writing is marked as a placeholder.

**Comment by sftim** [comment]
I'm going to mark this as WIP rather than “Placeholder” so that Prow understands that status.
If it's actually ready to review, feel free to re-retitle it
/retitle [WIP] Revise HPA documentation for stable HPA v2 API

**Comment by josephburnett** [comment]
Yes, it's a placeholder.  I'll be making some changes tomorrow so it will be ready for review.  Thanks!

File: content/en/docs/tasks/run-application/horizontal-pod-autoscale.md [hunk 1]
```diff
@@ -166,7 +166,7 @@ The current stable version, which only includes support for CPU autoscaling,
 can be found in the `autoscaling/v1` API version.
 
 The beta version, which includes support for scaling on memory and custom metrics,
```
[the remark this answers is not in the document]
Comment (josephburnett) [reply content/en/docs/tasks/run-application/horizontal-pod-autoscale.md:168]: Fixed.

File: content/en/docs/tasks/run-application/horizontal-pod-autoscale.md [hunk 2]
```diff
@@ -349,7 +349,7 @@ and [the walkthrough for using external metrics](/docs/tasks/run-application/hor
 
 Starting from
 [v1.18](https://github.com/kubernetes/enhancements/blob/master/keps/sig-autoscaling/853-configurable-hpa-scale-velocity/README.md)
-the `v2beta2` API allows scaling behavior to be configured through the HPA
+the `v2` API allows scaling behavior to be configured through the HPA
```
[the remark this answers is not in the document]
Comment (josephburnett) [reply content/en/docs/tasks/run-application/horizontal-pod-autoscale.md:352]: It's available in `v2beta2` from 1.18.  I added a note that it's also available from 1.23 in `v2`.

File: content/en/docs/tasks/run-application/horizontal-pod-autoscale.md [hunk 3]
```diff
@@ -314,7 +314,7 @@ annotations for specifying which custom metrics to scale on are no longer honore
 {{< /note >}}
 
 Kubernetes 1.6 adds support for making use of custom metrics in the Horizontal Pod Autoscaler.
```
[the remark this answers is not in the document]
Comment (josephburnett) [reply content/en/docs/tasks/run-application/horizontal-pod-autoscale.md:316]: Accepted.  Thanks.

File: content/en/docs/tasks/run-application/horizontal-pod-autoscale-walkthrough.md [hunk 4]
```diff
@@ -195,7 +195,7 @@ kubectl get hpa php-apache -o yaml > /tmp/hpa-v2.yaml
 Open the `/tmp/hpa-v2.yaml` file in an editor, and you should see YAML which looks like this:
 
 ```yaml
```
[the remark this answers is not in the document]
Comment (josephburnett) [reply content/en/docs/tasks/run-application/horizontal-pod-autoscale-walkthrough.md:197]: Yeah, how can I convert it?

**Comment by josephburnett** [comment]
This is ready for review now.  @sftim can you tell me how to convert that downloadable example?

**Comment by sftim** [comment]
Also remember to update https://kubernetes.io/docs/reference/using-api/deprecation-guide/ to mark the v2 beta APIs as deprecated.

File: content/en/docs/tasks/run-application/horizontal-pod-autoscale-walkthrough.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (sftim) [reply content/en/docs/tasks/run-application/horizontal-pod-autoscale-walkthrough.md:197]: You need to place a new file in [`examples`](https://github.com/kubernetes/website/tree/main/content/en/examples) - I'd put it in https://github.com/kubernetes/website/tree/main/content/en/examples/application/hpa 

and then use a `codenew` shortcode to reference it, similar to https://github.com/kubernetes/website/blame/1c31b83f9dba802ba5a45a1b83c4d4d220a40d4f/content/en/docs/tasks/run-application/horizontal-pod-autoscale-walkthrough.md#L65

File: content/en/docs/tasks/run-application/horizontal-pod-autoscale-walkthrough.md [hunk 5]
```diff
@@ -184,9 +184,9 @@ Autoscaling the replicas may take a few minutes.
...
 
 You can introduce additional metrics to use when autoscaling the `php-apache` Deployment
-by making use of the `autoscaling/v2beta2` API version.
+by making use of the `autoscaling/v2` API version.
 
-First, get the YAML of your HorizontalPodAutoscaler in the `autoscaling/v2beta2` form:
+First, get the YAML of your HorizontalPodAutoscaler in the `autoscaling/v2` form:
 
 ```shell
 kubectl get hpa php-apache -o yaml > /tmp/hpa-v2.yaml
```
Comment (sftim) [inline content/en/docs/tasks/run-application/horizontal-pod-autoscale-walkthrough.md:192]: nit: we have `kubectl edit` now

**Comment by sftim** [comment]
Changes look OK to me. **Separately**, the HPA docs [need](https://github.com/kubernetes/website/issues/4493) a good tidying.

@kubernetes/sig-autoscaling-pr-reviews are these changes technically accurate?

**Comment by reylejano** [comment]
@kubernetes/sig-autoscaling-misc , @kubernetes/sig-autoscaling-api-reviews ,  @gjtempleton 

The 1.23 Release Team requests a technical review, are the changes in this PR technically correct?

**Comment by gjtempleton** [comment]
> The 1.23 Release Team requests a technical review, are the changes in this PR technically correct?

This looks good to me.

**Pull request merged**